SwiftCSV/Parser.swift

```diff
@@ -118,12 +122,16 @@ enum Parser {
             }
 
             if !fields.isEmpty {
-                rowCallback(fields)
+                do {
+                    try rowCallback(fields)
+                } catch {
+                    throw CSVParseError.generic(message: "Error thrown by block for row \(rowIndex)")
+                }
```

Comment:
Same as above here, please: 

```suggestion
                try rowCallback(fields)
```